zbus/src/match_rule.rs

```diff
@@ -276,8 +284,8 @@ impl<'m> MatchRule<'m> {
         }
 
         // The arg0 namespace.
-        if let Some(arg0_ns) = self.arg0namespace() {
-            if let Ok(arg0) = msg.body_unchecked::<InterfaceName<'_>>() {
+        if let Some(arg0_ns) = self.arg0ns() {
+            if let Ok(arg0) = msg.body_unchecked::<Str<'_>>() {
                 if !arg0.starts_with(arg0_ns.as_str()) {
                     return Ok(false);
                 }
```

Comment:
- I believe this implementation was incorrect, because to match `arg0namespace`, the first N elements must match the namespace. For example, `arg0namespace='o'` should not match `org.foo` (just checked in using busctl monitor --match`). I guess it's better to fix in a new commit?
- I'm not sure, should it be `msg.body_unchecked::<BusName<'_>>()` instead?

Reply:
> * I believe this implementation was incorrect, because to match `arg0namespace`, the first N elements must match the namespace. For example, `arg0namespace='o'` should not match `org.foo` (just checked in using `busctl monitor --match`).

Nice catch. 

> I guess it's better to fix in a new commit?

You bet!

> I'm not sure, should it be `msg.body_unchecked::<BusName<'_>>()` instead?

I'm not sure either if we need to be strict. Let's make it stricter and if that's too strict, we can always revert back to `Str`.